src/storage/page_buffer.h

```diff
@@ -468,6 +468,8 @@ extern void pgbuf_daemons_destroy ();
 // Check if page is ahead of replication; only relevant on passive transaction server, don't call elsewhere.
 extern int pgbuf_check_page_ahead_of_replication (THREAD_ENTRY * thread_p, PAGE_PTR page);
```

Comment:
Activate pgbuf_check_page_ahead_of_replication declaration/definition also in not SERVER_MODE, to avoid putting #ifdef SERVER_MODE/#endif wherever it is being used.

Change its definition to:
```c
int
pgbuf_check_page_ahead_of_replication  (...)
{
#if defined (SERVER_MODE)
  // current body
#else
  return NO_ERROR;
#endif
}
```

Do something similar for pgbuf_check_for_deallocated_page_or_desyncronization:
``` c
int
pgbuf_check_for_deallocated_page_or_desyncronization (...)
{
  assert (err_errid () == ER_PB_BAD_PAGEID);
  ret = ER_PB_BAD_PAGEID;
#if defined (SERVER_MODE)
  // rest of current body
#else
  return ret;
#endif
}
```